src/calls.c

```diff
@@ -2006,17 +1993,17 @@ Obj FuncHandlerCookieOfFunction(Obj self, Obj func)
 */
 void SaveFunction ( Obj func )
 {
-  UInt i;
-  for (i = 0; i <= 7; i++)
-    SaveHandler(HDLR_FUNC(func,i));
-  SaveSubObj(NAME_FUNC(func));
```

Comment:
I had a look and you are right, if GCing happened during loading bad things would happen.